lib/core/Blockchain.ts

```diff
@@ -161,4 +164,33 @@ export default class Blockchain implements IBlockchain {
     console.info(`Refreshed blockchain time: ${responseBodyString}`);
     return responseBody;
   }
+
+  /**
+   * Fetches the normalized transaction fee used for proof-of-fee calculation, given the blockchain time.
+   * @param transactionNumber A valid Sidetree transaction number.
+   *
+   * @throws SidetreeError with ErrorCode.BlockchainTimeOutOfRange if the input transaction number is less
+   * than Sidetree genesis blockchain time or is later than the current blockchain time.
+   */
+  public async getFee (transactionNumber: number): Promise<number> {
+
+    const readUri = `${this.feeUri}/${transactionNumber}`;
+
+    const response = await this.fetch(readUri);
+    const responseBodyString = await ReadableStream.readAll(response.body);
+    const responseBody = JSON.parse(responseBodyString.toString());
+
+    if (response.status === HttpStatus.BAD_REQUEST &&
+        responseBody.code === SharedErrorCode.BlockchainTimeOutOfRange) {
+      throw new SidetreeError(SharedErrorCode.BlockchainTimeOutOfRange);
+    }
+
+    if (response.status !== HttpStatus.OK) {
+      console.error(`Blockchain read error response status: ${response.status}`);
+      console.error(`Blockchain read error body: ${responseBodyString}`);
+      throw new SidetreeError(CoreErrorCode.BlockchainReadResponseNotOk);
```

Comment:
Design: Rule of thumb adopted here is to not reuse error code. Consider creating a new error code.

Reply:
Oops .. thought that the code was generic. Good catch.